packages/turbo/ios/RNVisitableView.swift

```diff
@@ -34,72 +65,64 @@ class RNVisitableView: UIView, SessionSubscriber {
   required init?(coder aDecoder: NSCoder) {
     super.init(coder: aDecoder)
   }
-  
-  public func getRNSesssion() -> RNSession? {
-    if (!Thread.isMainThread) {
-      print("getSession accessed from incorrect thread")
-      return nil
-    }
     
-    guard let sessionModule = self.bridge?.uiManager.moduleRegistry.module(forName: "RNVisitableViewModule") as? RNVisitableViewModule else {
-        print("Couldn't find session for sessionHandle:", sessionHandle ?? "default session")
-        return nil
-    }
-    return sessionModule.getSession(sessionHandle: sessionHandle)
-  }
-  
-  public func getSession() -> Session? {
-    return getRNSesssion()?.turboSession
+  public func getWebViewConfiguration() -> WKWebViewConfiguration{
+    let configuration = WKWebViewConfiguration()
+    configuration.applicationNameForUserAgent = applicationNameForUserAgent as String?
+    return configuration
   }
   
   override func didMoveToWindow() {
-
-    let url = URL(string: String(url))!
-    
-    if (self.controller == nil) {
-      print("Open new VistableView with URL passed from JS: ", url)
-      setupViewController(url: url)
-      self.addSubview((controller?.view)!)
+    if (!subviews.contains(controller.view)){
+      addSubview(controller.view)
     }
   }
   
-  func setupViewController(url: URL) {
-    self.controller = RNVisitableViewController(url: url)
-    self.controller?.delegate = self
-    self.reactViewController().addChild(self.controller!)
-    self.controller?.view.frame = bounds // Fixes incorrect size of the webview
-    self.controller?.didMove(toParent: self.reactViewController())
-  }
-  
   func attachDelegateAndVisit(_ visitable: Visitable) {
-    let session = getSession()
-    session?.delegate = self
+    turboSession.delegate = self
     // Upon initial load, the session.visit function is called
     // to set the currentVisit private variable.
-    if(session?.webView.url == nil){
-      session?.visit(visitable)
+    if(webView.url == nil){
+      turboSession.visit(visitable)
       return
     }
-    session?.visitableViewWillAppear(visitable)
+    turboSession.visitableViewWillAppear(visitable)
+  }
+    
+  public func handleMessage(message: WKScriptMessage){
+    if let messageBody = message.body as? [AnyHashable : Any] {
+      onMessage?(messageBody)
+    }
+  }
+  
+  public func injectJavaScript(code: NSString) -> Void {
+    webView.evaluateJavaScript(code as String)
+  }
+    
+  private func sendSessionConfigurationWarning(propertyName: NSString){
+    onWarning?(["message": "You cannot change \(propertyName) after initialization of the session."])
+  }
+  
+  private func validateSessionProperty(propertyName: NSString, oldValue: NSString?, newValue: NSString?) -> Void{
+    if (oldValue != nil && newValue != nil  && !oldValue!.isEqual(to: newValue! as String)){
+      sendSessionConfigurationWarning(propertyName: propertyName)
+    }
```

Comment:
I'm inclined to say that this is a premature optimization having to maintain these (and also the corresponding `Session#configurationEquals` method. See also the comment in the `lazy var session` block.